components/script/dom/subtlecrypto.rs

```diff
@@ -115,6 +132,104 @@ impl SubtleCrypto {
 }
 
 impl SubtleCryptoMethods for SubtleCrypto {
+    /// <https://w3c.github.io/webcrypto/#SubtleCrypto-method-encrypt>
+    fn Encrypt(
```

Comment:
Encrypt and Decrypt share a lot of code. Can we create a function process_crypto_operation and call it from both Encrypt and Decrypt by passing an enum, something like this?

```
enum CryptoOperation {
    Encrypt,
    Decrypt,
}

self.process_crypto_operation(cx, algorithm, key, data, comp, CryptoOperation::Decrypt);
self.process_crypto_operation(cx, algorithm, key, data, comp, CryptoOperation::Encrypt);

```

Reply:
I did consider this yeah, my only concern is the slight variations between task pair requirements (encrypt/decrypt vs importKey/exportKey) complicating making a shared function. I'll take a look today though, will see if I can figure something out